dbg.h

```diff
@@ -35,25 +35,81 @@ License (MIT):
 #include <iostream>
 #include <sstream>
 #include <string>
+#include <type_traits>
 #include <vector>
```

Comment:
We should be able to remove the <vector> include.

Reply:
`std::enable_if`?

Reply:
Not `type_traits`, the `vector` include.